File: leetcode/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indexMap = dict()
        
        for i in range(len(nums)):
            index = indexMap.get(target - nums[i], -1)
            if index != -1:
                ans = list()
                ans.append(index)
                ans.append(i)
                return ans
            
            indexMap[nums[i]] = i
        
        return None

File: leetcode/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> indexMap;
        
        vector<int> ans(2);
        for (int i = 0; i < nums.size(); i++) {
            if (indexMap.find(target - nums[i]) != indexMap.end()) {
                ans[0] = indexMap[target - nums[i]];
                ans[1] = i;
                return ans;
            }
            
            indexMap[nums[i]] = i;
        }
        
        return ans;
    }
};